packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -1081,21 +1099,54 @@ export class DatabaseCluster extends DatabaseClusterNew {
   constructor(scope: Construct, id: string, props: DatabaseClusterProps) {
     super(scope, id, props);
 
-    const credentials = renderCredentials(this, props.engine, props.credentials);
-    const secret = credentials.secret;
+    if (
+      props.manageMasterUserPassword &&
+      (
+        props.credentials?.excludeCharacters ||
+        props.credentials?.password ||
+        props.credentials?.replicaRegions ||
+        props.credentials?.secret ||
+        props.credentials?.secretName ||
+        props.credentials?.usernameAsString
+      )
+    ) {
+      throw new Error('Only the `username` and `encryptionKey` credentials properties may be used when `manageMasterUserPassword` is true');
+    }
 
-    const cluster = new CfnDBCluster(this, 'Resource', {
-      ...this.newCfnProps,
-      // Admin
-      masterUsername: credentials.username,
-      masterUserPassword: credentials.password?.unsafeUnwrap(),
-    });
+    let cluster: CfnDBCluster;
+    if (!props.manageMasterUserPassword) {
+      const credentials = renderCredentials(this, props.engine, props.credentials);
+      const secret = credentials.secret;
 
-    this.clusterIdentifier = cluster.ref;
-    this.clusterResourceIdentifier = cluster.attrDbClusterResourceId;
+      cluster = new CfnDBCluster(this, 'Resource', {
+        ...this.newCfnProps,
+        // Admin
+        masterUsername: credentials.username,
+        masterUserPassword: credentials.password?.unsafeUnwrap(),
+      });
+
+      this.clusterIdentifier = cluster.ref;
+      this.clusterResourceIdentifier = cluster.attrDbClusterResourceId;
+
+      if (secret) {
+        this.secret = secret.attach(this);
+      }
+    } else {
+      cluster = new CfnDBCluster(this, 'Resource', {
+        ...this.newCfnProps,
+        // Admin
+        // Don't set password if RDS is managing credentials.
+        masterUsername: props.credentials?.username ?? Credentials.fromUsername(props.engine.defaultUsername ?? 'admin').username,
+        // Define kms key the RDS Managed Secret will use.
+        ...props.credentials?.encryptionKey !== undefined && {
+          masterUserSecret: {
```

Comment:
Shouldn't `this.secret` be set by `CfnDBCluster`'s `masterUserSecret.secretArn`? I don't see a direct way to access the created secret for use within CDK.

Reply:
There wouldn't be one unfortunately. I'm not sure Cfn has a way to reference it either. The AWS SDK does allow you to get the secret's ARN by describing the cluster.

Reply:
Not sure if it helps, but this is how I was getting access to the secret so that I could include it in the task definition for my ECS tasks that are created by the same stack.

`((CfnDBCluster)cfnCluster.Node.DefaultChild).AttrMasterUserSecretSecretArn`

Reply:
@antonfelich Thanks for the comment! After a bit of testing I was able to use a similar technique to create an imported secret that's accessible from the cluster's `secret` prop. Should be a great addition!

Line 1169 in cluster.ts:
`this.secret = secretsmanager.Secret.fromSecretCompleteArn(this, 'DbClusterSecret', cluster.attrMasterUserSecretSecretArn)`